back-end-side/Controllers/ShowDataController.cs

```diff
@@ -76,18 +78,18 @@ public IActionResult Put([FromBody] Record model, int id)
         }
         [HttpDelete("{id:int}")]
         [EnableCors("corsapp")]
-        public IActionResult Delete(int id)
+        public async Task<IActionResult> Delete(int id)
         {
-            var record = _context.Expenses.AsNoTracking().FirstOrDefault(r => r.ID.Equals(id));
+            var record = await _context.Expenses.FindAsync(id);
             if (record == null)
             {
-                return NotFound();
+                Console.WriteLine("Couldn't find record");
```

Comment:
is this necessary?

Reply:
No, I left it here as a temporary solution, we will have to log the errors in a more meaningful way later